android-agent/src/main/java/io/opentelemetry/android/SessionIdTimeoutHandler.java

```diff
@@ -25,19 +25,24 @@
  */
 final class SessionIdTimeoutHandler implements ApplicationStateListener {
 
-    private static final long SESSION_TIMEOUT_NANOS = TimeUnit.MINUTES.toNanos(15);
+    private final long sessionTimeoutNanos;
 
     private final Clock clock;
     private volatile long timeoutStartNanos;
     private volatile State state = State.FOREGROUND;
 
     SessionIdTimeoutHandler() {
-        this(Clock.getDefault());
+        this(Clock.getDefault(), TimeUnit.MINUTES.toNanos(15));
+    }
+
+    SessionIdTimeoutHandler(long sessionTimeoutMinutes) {
```

Comment:
Any reason we don't use `Duration` here as well to make as more flexible? A granularity of minutes seems a bit coarse for general usage. Doing that will also mean we don't have to add a separate method just for testing, reducing the API surface.

Reply:
Agreed, cleaner to use Duration here as well.